sdlib/d/gc/slab.d

```diff
@@ -98,6 +98,53 @@ unittest SlabRange {
 	assert(r0.length == 12);
 }
 
+struct SlotMetadata {
+private:
+	struct FreeSpaceData {
+		private ushort[size_t.sizeof / ushort.sizeof - 1] _pad1;
+		ushort freeSpace;
+	}
+
+	union Data {
+		size_t finalizerData;
+		FreeSpaceData freeSpaceData;
+	}
+
+	static assert(Data.sizeof == size_t.sizeof);
+
+	Data data;
+
+	enum FinalizerBit = nativeToBigEndian!size_t(0x2);
+
+public:
+	@property
+	ushort freeSpace() {
+		return readPackedFreeSpace(&data.freeSpaceData.freeSpace);
+	}
+
+	void setFreeSpace(size_t size) {
+		assert(size > 0, "Attempt to set a slot metadata size of 0!");
+
+		writePackedFreeSpace(&data.freeSpaceData.freeSpace, size);
+	}
+
+	@property
+	Finalizer finalizer() {
+		return (data.finalizerData & FinalizerBit)
```

Comment:
hasFinalizer

Reply:
Not sure if it's important, but hasFinalizer converts to a bool, whereas this just checks for the bit. But I can switch it back.